sdk/src/wallet/account/operations/transaction/prepare_output.rs

```diff
@@ -97,263 +96,182 @@ where
+                let new_amount = params.amount + min_storage_deposit_basic_output;
+                // new_amount could be not enough because we added the storage deposit return unlock condition, so we
+                // need to check the min required storage deposit again
+                let min_storage_deposit_new_amount = second_output_builder
+                    .clone()
+                    .with_minimum_storage_deposit(rent_structure)
+                    .finish_output(token_supply)?
+                    .amount();
+
+                if new_amount < min_storage_deposit_new_amount {
+                    let additional_required_amount = min_storage_deposit_new_amount - new_amount;
+                    second_output_builder = second_output_builder.with_amount(new_amount + additional_required_amount);
+                    // Add the additional amount to the SDR
                     second_output_builder =
-                        second_output_builder.add_unlock_condition(StorageDepositReturnUnlockCondition::new(
+                        second_output_builder.replace_unlock_condition(StorageDepositReturnUnlockCondition::new(
                             remainder_address,
                             // Return minimum storage deposit
-                            min_storage_deposit_return_amount,
+                            min_storage_deposit_basic_output + additional_required_amount,
                             token_supply,
                         )?);
+                } else {
+                    // new_amount is enough
+                    second_output_builder = second_output_builder.with_amount(new_amount);
                 }
-
-                // Check if the remainder balance wouldn't leave dust behind, which wouldn't allow the creation of this
-                // output. If that's the case, this remaining amount will be added to the output, to still allow sending
-                // it.
-                if storage_deposit.use_excess_if_low.unwrap_or_default() {
-                    let balance = self.balance().await?;
-                    if balance.base_coin.available.cmp(&first_output.amount()) == Ordering::Greater {
-                        let balance_minus_output = balance.base_coin.available - first_output.amount();
-                        // Calculate the amount for a basic output
-                        let minimum_required_storage_deposit =
-                            MinimumStorageDepositBasicOutput::new(rent_structure, token_supply).finish()?;
-
-                        if balance_minus_output < minimum_required_storage_deposit {
-                            second_output_builder =
-                                second_output_builder.with_amount(first_output.amount() + balance_minus_output);
-                        }
-                    }
-                }
+                // Set a bool to contains_return = true so it can be increased later if needed? Or just get mut
```

Comment:
?

Reply:
Comment I wrote for myself when I thought about how to do it, removed